### PR TITLE
Add support for declarative stimulus/reflex behavior

### PR DESCRIPTION
**Pull request by hopsoft**
Add declarative stimulus/reflex behavior via data attributes.

- `data-reflex` - accepts a Stimulus styled value to specify DOM event & server side reflexes i.e. `click->ExampleReflex#method`

Very much inspired by #12

Example usage here: https://github.com/hopsoft/stimulus_reflex_todomvc/pull/16

I have a question out to the Stimulus team to determine if the techniques used here are valid. i.e. calling `application.start` inside a library. https://github.com/stimulusjs/stimulus/issues/272

### CODE REVIEW

**Comment by hopsoft** [comment]
I'm curious what other's think about the use of 2 data attributes vs 1. The other proposal was a single data attribute that matches Stimulus semantics. See #12 for more context.

```html
<div data-reflex="click->ServerSideReflex#do_something">
```

-vs-

```html
<div data-stimulus="click" data-reflex="ServerSideReflex#do_something">
```

**Comment by hopsoft** [comment]
I'm starting to lean toward the single data attribute as initially proposed by @pjforde1978

File: javascript/yarn.lock [hunk 1]
```diff
@@ -2,6 +2,30 @@
 # yarn lockfile v1
 
 
+"@stimulus/core@^1.1.1":
```
Comment (andrewmcodes) [inline javascript/yarn.lock]: Shouldn't this reflect v2.0.0?

File: lib/stimulus_reflex/channel.rb [hunk 2]
```diff
@@ -22,11 +22,11 @@ def receive(data)
     reflex_name, method_name = target.split("#")
     reflex_name = reflex_name.classify
     arguments = data["args"] || []
-    options = StimulusReflex::DomElement.new(data["attrs"])
+    element = StimulusReflex::Element.new(data["attrs"])
```
Comment (andrewmcodes) [inline lib/stimulus_reflex/channel.rb]: Been thinking about this more and I still like element.

Seems inline with the [MDN docs](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/dataset)

File: javascript/yarn.lock (hunk 1 above)
Comment (pjforde1978) [reply javascript/yarn.lock]: Is Stimulus v2 imminent?

Comment (andrewmcodes) [reply javascript/yarn.lock]: @pjforde1978 the changes in this PR that Nate and I paired on are breaking so I’d say yes though it’s obviously ultimately up to Nate.

Comment (hopsoft) [reply javascript/yarn.lock]: That dep is for Stimulus itself rather than StimulusReflex.

**Pull request merged**